cmd/internal/register/maps.go

```diff
@@ -113,16 +131,47 @@ func Maps(a *atlas.Atlas, maps []config.Map, providers map[string]provider.Tiler
 
 			// confirm our providerLayer name is registered
 			var found bool
+			// this must be an array because auto providers may have multiple layers that match regex
+			var provLayers []string
 			var layerGeomType tegola.Geometry
-			for i := range layerInfos {
-				if layerInfos[i].Name() == providerLayer[1] {
-					found = true
+			for i, info := range layerInfos {
+				// check if provider layers are automatically generated
+				if auto {
```

Comment:
Don't know why we need this check.The expression: `^san_deigo` would still match `san_diego` as the name. So, why not just have two cases here:
```go

if providerLayer[1] == "*" {
  // return all the layers:
    provLayers = append(provLayers,info.Name())
    found = true // Do we need this now as len(provLayers) is the same?
    layerGeoType = info.GeoType()
    continue
} 
r, err := regex.Compile("^"+providerLayer[1])
if err != nil {
       log.Printf("Error when parsing regex (layer: %v): %v", info.Name(), err)
       continue Provider // add a Providers label at 111
}
if !r.MatchString(info.Name()) {
    continue
}
provLayers = append(provLayers, info.Name())
found = true
layerGeoType = info.GeoType()

```

Reply:
Can you explain this: `continue Provider // add a Providers label at 111`

I'm a golang newbie (mostly use JS) so not sure what you mean by Providers label etc